src/compiler.ts

```diff
@@ -8756,7 +8769,8 @@ export class Compiler extends DiagnosticEmitter {
             isWasm64
               ? module.i64(i64_low(bufferAddress), i64_high(bufferAddress))
               : module.i32(i64_low(bufferAddress))
-          ], expression)
+          ], expression),
+          isWasm64 ? Type.i64 : Type.i32
```

Comment:
This is retaining an instance of `ArrayBuffer` of type `allocBufferInstance.type`. Can then also pull `allocBufferInstance` into a local, so it looks like

```ts
let allocBufferInstance = program.allocBufferInstance;
let expr = this.makeRetain(
  this.makeCallDirect(allocBufferInstance, [
  ...
  allocBufferInstance.type
);
```

Reply:
A, wait, that's not right, going to update the comment above.

Reply:
For the closures PR it doesn't matter what this type is, since it only cares about types with signature references.

Agreed passing in the input type doesn't really make sense, as it would always just be a usize or equivalent to the output type. Will make that change